pyEpiabm/pyEpiabm/routine/file_population_config.py

```diff
@@ -96,13 +100,21 @@ def make_pop(input_file: str, random_seed: int = None, time: float = 0):
 
             # Raise error if microcell exists, then create new one
             for microcell in cell.microcells:
-                if microcell.id == line["microcell"]:
+
+                # The microcell id is a string separated by periods, so we
+                # must find the part which matches the csv input
+                id_parts = microcell.id.split(".")
+                if id_parts[-1] == microcell_id_csv:
                     raise ValueError(f"Duplicate microcells {microcell.id}"
                                      + f" in cell {cell.id}")
 
             new_microcell = Microcell(cell)
             cell.microcells.append(new_microcell)
-            new_microcell.set_id(line["microcell"])
+
+            # Either keep the following line (sets id according to data frame
+            # input), or remove it (sets id according to current number of
+            # microcells in cell.microcells)
```

Comment:
I don't understand this comment - are you suggesting we should comment the following line in some contexts? I would be against this, can we automate this different behavior in different contexts

Reply:
Apologies, I meant line 117 not 115, but after the discussions we've had I will remove this comment now - this issue has been sorted